AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -208,17 +205,20 @@ - (AWSTask *)saveRecord:(NSData *)data
                                    ];
                     if (!result) {
                         AWSLogError(@"SQLite error. [%@]", db.lastError);
-                        error = db.lastError;
-                        return;
+                        dbError = db.lastError;
+                        return [AWSTask taskWithError:dbError];
```

Comment:
I totally see it's a little confusing since the blocks are nested. Here is the definition of `- inDatabase:`

    - (void)inDatabase:(void (^)(AWSFMDatabase *db))block;

This is different than the `AWSTask` related blocks. This particular block needs to return `void`.